src/services/news.js

```diff
@@ -83,6 +83,37 @@ const getNotExpiredFormatted = async () => {
   return news;
 };
 
+/**
+ * Gets unexpired student news matching a search string
+ * @param {NewsItem[]} unexpiredNews the list of unfiltered news items
+ * @param {String} query the search query
+ * @return {Promise<any>} Student news
+ */
+const getFilteredNews = (unexpiredNews, query) => {
+  const filteredNews = [];
+  unexpiredNews.map((newsitem) => {
```

Comment:
Rather than creating and pushing to an intermediate array, you can use `Array.filter()` and return an array containing elements from `unexpiredNews` that return true in the code below.
```suggestion
  return unexpiredNews.map((newsitem) => {
```

Reply:
I'm not adding every element to the new array and I don't want to add NULLs, so can I really use map?

Reply:
Sorry, meant to change that to `unexpiredNews.filter`